modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Select/Select.es.js

```diff
@@ -12,449 +12,337 @@
+			)
+		)
+		.filter(({value}) => value !== '');
 
-	_handleDocumentClicked({target}) {
-		const {base} = this.refs;
-		const {dropdown} = base.refs;
-
-		if (dropdown) {
-			const {menu} = dropdown.refs.portal.refs;
-			const {expanded} = this;
-
-			if (
-				expanded &&
-				!this.element.contains(target) &&
-				!dropdown.element.contains(target) &&
-				!menu.contains(target)
-			) {
-				this.setState({
-					expanded: false,
-				});
-			}
-		}
+	if (!multiple) {
+		return [emptyOption, ...newOptions];
 	}
 
-	_handleExpandedChanged({newVal}) {
-		if (newVal) {
-			this.emit('fieldFocused', {
-				fieldInstance: this,
-				originalEvent: window.event,
-			});
-		}
-		else {
-			this.emit('fieldBlurred', {
-				fieldInstance: this,
-				originalEvent: window.event,
-			});
-		}
-
-		this.expanded = newVal;
-	}
+	return newOptions;
+}
 
-	_handleItemClicked({data, preventDefault}) {
-		const {multiple} = this;
-		const currentValue = this._getArrayValue(this.value);
-		const itemValue = data.item.value;
-
-		let newValue;
-
-		if (multiple) {
-			if (currentValue.includes(itemValue)) {
-				newValue = this.deleteValue(itemValue);
-
-				if (document.activeElement) {
-					document.activeElement.blur();
-				}
-			}
-			else {
-				newValue = this.addValue(itemValue);
-			}
-		}
-		else {
-			newValue = this.setValue(itemValue);
-		}
-
-		preventDefault();
-
-		this.setState(
-			{
-				expanded: multiple,
-				value: newValue,
-			},
-			() =>
-				this.emit('fieldEdited', {
-					fieldInstance: this,
-					value: newValue,
-				})
+const Trigger = forwardRef(
+	({onCloseButtonClicked, onTriggerClicked, value, ...otherProps}, ref) => {
+		return (
+			<>
+				<HiddenSelectInput value={value} {...otherProps} />
+				<VisibleSelectInput
+					onClick={onTriggerClicked}
+					onCloseButtonClicked={onCloseButtonClicked}
+					ref={ref}
+					value={value}
+					{...otherProps}
+				/>
+			</>
 		);
 	}
+);
+
+const Select = ({
+	multiple,
+	onCloseButtonClicked,
+	onDropdownItemClicked,
+	onExpand,
+	options,
+	predefinedValue,
+	readOnly,
+	value,
+	...otherProps
+}) => {
+	const menuElementRef = useRef(null);
+	const triggerElementRef = useRef(null);
+
+	const [currentValue, setCurrentValue] = useState(value);
+	const [expand, setExpand] = useState(false);
+
+	return (
+		<div>
+			<Trigger
+				onCloseButtonClicked={({event, value}) => {
+					const newValue = removeValue({
+						value: currentValue,
+						valueToBeRemoved: value,
+					});
+
+					setCurrentValue(newValue);
+
+					onCloseButtonClicked({event, value: newValue});
+				}}
+				onTriggerClicked={event => {
+					event.preventDefault();
+
+					if (readOnly) {
+						return;
+					}
+
+					triggerElementRef.current = event.target;
+
+					setExpand(!expand);
+					onExpand({event, expand: !expand});
+				}}
+				options={options}
+				predefinedValue={predefinedValue}
+				readOnly={readOnly}
+				value={currentValue}
+				{...otherProps}
+			/>
+			<ClayDropDown.Menu
```

Comment:
@diegonvs we will have to implement the focus manually here, you can check an example in Text